src/material-experimental/mdc-chips/chip-action.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  AfterViewInit,
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  Inject,
+  OnChanges,
+  OnDestroy,
+  SimpleChanges,
+} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {
+  MDCChipActionAdapter,
+  MDCChipActionFoundation,
+  MDCChipActionType,
+  MDCChipPrimaryActionFoundation,
+} from '@material/chips';
+import {emitCustomEvent} from './emit-event';
+import {
+  CanDisable,
+  HasTabIndex,
+  mixinDisabled,
+  mixinTabIndex,
+} from '@angular/material-experimental/mdc-core';
+import {BooleanInput, NumberInput} from '@angular/cdk/coercion';
+
+const _MatChipActionMixinBase = mixinTabIndex(mixinDisabled(class {}), -1);
+
+/** @docs-private */
```

Comment:
Even though this is private, could you add a JsDoc class description for anyone working on the chips?